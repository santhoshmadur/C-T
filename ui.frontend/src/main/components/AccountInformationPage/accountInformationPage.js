/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React, { Fragment, Suspense, useEffect } from 'react';
import { FormattedMessage } from 'react-intl';
import { useHistory } from 'react-router-dom';
import { useAccountInformationPage } from '@magento/peregrine/lib/talons/AccountInformationPage/useAccountInformationPage';
import { useConfigContext } from '@adobe/aem-core-cif-react-components';

import { useStyle } from '@magento/venia-ui/lib/classify';
import Button from '@magento/venia-ui/lib/components/Button';
import { Message } from '@magento/venia-ui/lib/components/Field';
import { fullPageLoadingIndicator } from '@magento/venia-ui/lib/components/LoadingIndicator';

import defaultClasses from '@magento/venia-ui/lib/components/AccountInformationPage/accountInformationPage.css';
import AccountInformationPageOperations from '@magento/venia-ui/lib/components/AccountInformationPage/accountInformationPage.gql.js';

const EditModal = React.lazy(() => import('@magento/venia-ui/lib/components/AccountInformationPage/editModal'));

const AccountInformationPage = props => {
    const { pagePaths } = useConfigContext();
    const classes = useStyle(defaultClasses, props.classes);

    const history = useHistory();
    const talonProps = useAccountInformationPage({
        ...AccountInformationPageOperations
    });

    const {
        handleCancel,
        formErrors,
        handleChangePassword,
        handleSubmit,
        initialValues,
        isDisabled,
        isSignedIn,
        isUpdateMode,
        loadDataError,
        shouldShowNewPassword,
        showUpdateMode
    } = talonProps;

    // If the user is no longer signed in, redirect to the home page.
    useEffect(() => {
        if (!isSignedIn) {
            history.replace(pagePaths.baseUrl);
            history.go(0);
        }
    }, [history, isSignedIn]);

    const errorMessage = loadDataError ? (
        <Message>
            <FormattedMessage
                id={'accountInformationPage.errorTryAgain'}
                defaultMessage={'Something went wrong. Please refresh and try again.'}
            />
        </Message>
    ) : null;

    let pageContent = null;
    if (!initialValues) {
        return fullPageLoadingIndicator;
    } else {
        const { customer } = initialValues;
        const customerName = `${customer.firstname} ${customer.lastname}`;
        const passwordValue = '***********';

        pageContent = (
            <Fragment>
                <div className={classes.accountDetails}>
                    <div className={classes.lineItemsContainer}>
                        <span className={classes.nameLabel}>
                            <FormattedMessage id={'global.name'} defaultMessage={'Name'} />
                        </span>
                        <span className={classes.nameValue}>{customerName}</span>
                        <span className={classes.emailLabel}>
                            <FormattedMessage id={'global.email'} defaultMessage={'Email'} />
                        </span>
                        <span className={classes.emailValue}>{customer.email}</span>
                        <span className={classes.passwordLabel}>
                            <FormattedMessage id={'global.password'} defaultMessage={'Password'} />
                        </span>
                        <span className={classes.passwordValue}>{passwordValue}</span>
                    </div>
                    <div className={classes.editButtonContainer}>
                        <Button
                            className={classes.editInformationButton}
                            disabled={false}
                            onClick={showUpdateMode}
                            priority="normal"
                        >
                            <FormattedMessage id={'global.editButton'} defaultMessage={'Edit'} />
                        </Button>
                    </div>
                </div>
                <Suspense fallback={null}>
                    <EditModal
                        formErrors={formErrors}
                        initialValues={customer}
                        isDisabled={isDisabled}
                        isOpen={isUpdateMode}
                        onCancel={handleCancel}
                        onChangePassword={handleChangePassword}
                        onSubmit={handleSubmit}
                        shouldShowNewPassword={shouldShowNewPassword}
                    />
                </Suspense>
            </Fragment>
        );
    }

    return (
        <div className={classes.root}>
            <h1 className={classes.title}>
                <FormattedMessage
                    id={'accountInformationPage.accountInformation'}
                    defaultMessage={'Account Information'}
                />
            </h1>
            {errorMessage ? errorMessage : pageContent}
        </div>
    );
};

export default AccountInformationPage;
