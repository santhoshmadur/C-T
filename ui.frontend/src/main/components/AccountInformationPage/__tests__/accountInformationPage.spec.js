/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import render from '../../utils/test-utils';
import { useAccountInformationPage } from '@magento/peregrine/lib/talons/AccountInformationPage/useAccountInformationPage';
import AccountInformationPage from '../accountInformationPage';
import LoadingIndicator from '@magento/venia-ui/lib/components/LoadingIndicator';
import { ConfigContextProvider } from '@adobe/aem-core-cif-react-components';
import { useHistory } from 'react-router-dom';

jest.mock('@magento/peregrine/lib/talons/AccountInformationPage/useAccountInformationPage');
jest.mock('@magento/venia-ui/lib/classify');

const handleSubmit = jest.fn().mockName('handleSubmit');
const handleCancel = jest.fn().mockName('handleCancel');
const showUpdateMode = jest.fn().mockName('showUpdateMode');

const emptyFormProps = {
    handleCancel,
    formErrors: [],
    handleSubmit,
    initialValues: {
        customer: {
            firtname: 'Foo',
            lastname: 'Bar',
            email: 'foobar@express.net'
        }
    },
    isDisabled: false,
    isSignedIn: true,
    isUpdateMode: false,
    loadDataError: null,
    showUpdateMode
};

const config = {
    pagePaths: {
        baseUrl: '/content/venia/us/en.html'
    }
};

jest.mock('react-router-dom', () => {
    const replace = jest.fn();
    const go = jest.fn();
    return {
        useHistory: () => ({
            replace,
            go
        })
    };
});

describe('AccountInformationPage', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('redirects when not authenticated', () => {
        useAccountInformationPage.mockReturnValue({
            isSignedIn: false
        });

        const { replace, go } = useHistory();

        render(
            <ConfigContextProvider config={config}>
                <AccountInformationPage />
            </ConfigContextProvider>
        );
        expect(replace).toHaveBeenCalledTimes(1);
        expect(replace).toHaveBeenCalledWith(config.pagePaths.baseUrl);
        expect(go).toHaveBeenCalledTimes(1);
        expect(go).toHaveBeenCalledWith(0);
    });

    test('renders a loading indicator', () => {
        useAccountInformationPage.mockReturnValueOnce({
            initialValues: null,
            isSignedIn: true
        });

        const { root } = render(
            <ConfigContextProvider config={config}>
                <AccountInformationPage />
            </ConfigContextProvider>
        );

        expect(root.findByType(LoadingIndicator)).toBeTruthy();
    });

    test('renders form error', () => {
        useAccountInformationPage.mockReturnValueOnce({
            ...emptyFormProps,
            loadDataError: { loadDataError: 'Form Error' }
        });

        const tree = render(
            <ConfigContextProvider config={config}>
                <AccountInformationPage />
            </ConfigContextProvider>
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
