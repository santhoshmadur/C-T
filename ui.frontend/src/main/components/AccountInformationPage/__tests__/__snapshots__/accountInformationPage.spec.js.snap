// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInformationPage renders form error 1`] = `
<div
  className="root"
>
  <h1
    className="title"
  >
    Account Information
  </h1>
  <p
    className="root"
  >
    Something went wrong. Please refresh and try again.
  </p>
</div>
`;
